### PR TITLE
GH-15202: [R] Function create_package_with_all_dependencies fails

### LINKED ISSUE #15202: [R] Function create_package_with_all_dependencies fails
### Describe the bug, including details regarding any error messages, version, and platform.

**Platform** R
**R version** 3.5.1
**Arrow version** arrow_10.0.1

The `create_package_with_all_dependencies` function fails with the following error message

`Error in download.file("https://sourceware.org/pub/bzip2/bzip2-1.0.8.tar.gz",  : `
`  cannot open URL 'https://sourceware.org/pub/bzip2/bzip2-1.0.8.tar.gz'`
`In addition: Warning message:`
`In download.file("https://sourceware.org/pub/bzip2/bzip2-1.0.8.tar.gz",  :`
` Show Traceback`
` Rerun with Debug`
` Error in download.file("https://sourceware.org/pub/bzip2/bzip2-1.0.8.tar.gz",  : `
 ` cannot open URL 'https://sourceware.org/pub/bzip2/bzip2-1.0.8.tar.gz'`

The solution is to run the following command before to avoid missing CA certificates of https-links
`options(download.file.method="curl", download.file.extra="-k -L")`

### Component(s)

R

### PR DESCRIPTION
**Pull request by thisisnic**
The linked user issue shows a failure of `create_package_with_all_dependencies()` due to a problem downloading a dependency, the nature of which is unclear from the output. If we allow output shown when downloading these dependencies, it'll make it easier for users to diagnose issues when things go wrong.

(We might also not want to merge this as this was an unusual situation, and we might not want that much output to be displayed?)
* Closes: #15202

### CODE REVIEW

File: r/tools/download_dependencies_R.sh [hunk 1]
```diff
@@ -45,7 +45,7 @@ download_dependency() {
   local url=$1
   local out=$2
 
-  echo 'download.file("'${url}'", "'${out}'", quiet = TRUE)'
```
[the remark this answers is not in the document]
Comment (thisisnic) [reply r/tools/download_dependencies_R.sh:48]: Great idea!

**Comment by assignUser** [comment]
Regarding the cert issue I would also be in favor of changing that option (ideally only temporarily with a reset `on.exit`)

**Comment by thisisnic** [comment]
> Regarding the cert issue I would also be in favor of changing that option (ideally only temporarily with a reset `on.exit`)

Isn't that a security issue?  If it's not, I'd be in favour of that change instead of this one.

**Comment by assignUser** [comment]
Oh yeah true: https://curl.se/docs/manpage.html#-k
Maybe just a note in the docs that this is a possible workaround (so that we don't apply it by default)?

**Comment by thisisnic** [comment]
@paleolimbot I might be misremembering, but I believe that we briefly discussed this issue last week and you had an alternative/additional solution to suggest here? I can't recall what it was, do you know?

**Comment by thisisnic** [comment]
I'm a bit hesitant to document a workaround, @assignUser, as this feels like a non-Arrow-related problem, which isn't super-common.

Perhaps this should just be closed without any changes being made?  I'm not sure doing anything actually improves things here.

**Comment by assignUser** [comment]
@thisisnic yes I agree, I was missing the outdated env of the user. This is therefore likely a one off issue/not-arrow and we can close this PR (and the issue, which serves as documentation in itself).

**Pull request closed without merging**